src/NuGet.Core/NuGet.Packaging/Signing/SignatureVerificationProviderFactory.cs

```diff
@@ -7,12 +7,13 @@ namespace NuGet.Packaging.Signing
 {
     public static class SignatureVerificationProviderFactory
     {
-        public static IEnumerable<ISignatureVerificationProvider> GetSignatureVerificationProviders()
+        public static IEnumerable<ISignatureVerificationProvider> GetSignatureVerificationProviders(SignatureVerificationProviderArgs args)
```

Comment:
Consider the following - 
```
public static class SignatureVerificationProviderFactory
{
    public static IEnumerable<ISignatureVerificationProvider> GetSignatureVerificationProviders(SignatureVerificationProviderArgs args)
    {
        if (args == null)
        {
            throw new ArgumentNullException(nameof(args))
        }
        
        return new List<ISignatureVerificationProvider>()
        {
            new IntegrityVerificationProvider(),
            new SignatureTrustAndValidityVerificationProvider(args.FingerprintAlgorithm),
            new WhitelistVerificationProvider(args.FingerprintAlgorithm, args.Whitelist)
        };
    }
    
    public static IEnumerable<ISignatureVerificationProvider> GetSignatureVerificationProviders()
    {        
        return new List<ISignatureVerificationProvider>()
        {
            new IntegrityVerificationProvider(),
            new SignatureTrustAndValidityVerificationProvider(HashAlgorithmName.SHA256)
        };
    }
}
```

This will prevent all those calls to `new SignatureVerificationProviderArgs ()` and calls into `WhitelistVerificationProvider` since we always have args.

Reply:
@mishra14 I had thought about this... I prefer making this change to now support adding parameters to any verification provider added. I don't believe we should do this change, because maybe in the future we would need more verification providers receiving a different combination of arguments and can become confusing.